connection.go

```diff
@@ -69,11 +69,14 @@ type ConnectionBuilder struct {
 	password     string
 	tokens       []string
 	scopes       []string
+	transport    WrappedTransport
 
 	// Metrics:
 	subsystem string
 }
 
+type WrappedTransport = func(http.RoundTripper) http.RoundTripper
```

Comment:
Why a type alias and not just a function type?

Add documentation comments.

Reply:
Added